modules/Mail/Spamgourmet/Config.pm

```diff
@@ -287,84 +296,88 @@ sub getDBConnection {
 #   # die "$DBI::errstr\n";
 #  }
 
-  my $connectTries = 0;
-  my $interval = 6; # crisis mode
-  my $connected = 0;
-  my $totaltime = 0;
-  while (!$connected && $connectTries < 50) {
-    $connected = $self->{'db'} = DBI->connect($self->{'dbstring'},$self->{'dbuser'},$self->{'dbpassword'});
-    if (!$connected) {
-  $self->debug("failed attempt to connect - try number: $connectTries, total time: $totaltime");
-      sleep $interval;
-      $totaltime += $interval;
+    my $connectTries = 0;
+    my $interval     = 6;    # crisis mode
+    my $connected    = 0;
+    my $totaltime    = 0;
+    while ( !$connected && $connectTries < 50 ) {
+        $connected = $self->{'db'}
+            = DBI->connect( $self->{'dbstring'}, $self->{'dbuser'},
+            $self->{'dbpassword'} );
+        if ( !$connected ) {
+            $self->debug(
+                "failed attempt to connect - try number: $connectTries, total time: $totaltime"
+            );
+            sleep $interval;
+            $totaltime += $interval;
+        }
+        $connectTries++;
+        $interval = 15  if ( $connectTries > 10 );    # emergency mode
+        $interval = 120 if ( $connectTries > 20 );    # apocalypse mode
+    }
+    if ( !$connected ) {
+        $self->debug("could not connect to db: $DBI::errstr\n");
+        die EX_TEMPFAIL
+            if $self->{'mode'} == 0
+            ;   # if we're running under sendmail, we want to issue a TEMPFAIL
```

Comment:
As exemplified by these lines, the readability of the reformatted code is not always ideal but at least we have a normalized format now.